src/utils/web3/ata-utils.ts

```diff
@@ -23,41 +25,89 @@ import { createWSOLAccountInstructions } from "./token-instructions";
  * @returns
  */
 export async function resolveOrCreateATA(
+  dal: OrcaDAL,
   connection: Connection,
   ownerAddress: PublicKey,
   tokenMint: PublicKey,
   wrappedSolAmountIn = new u64(0)
 ): Promise<ResolvedTokenAddressInstruction> {
-  if (!tokenMint.equals(NATIVE_MINT)) {
-    const ataAddress = await deriveATA(ownerAddress, tokenMint);
+  const instructions = await resolveOrCreateATAs(dal, connection, ownerAddress, [
+    { tokenMint, wrappedSolAmountIn },
+  ]);
+  return instructions[0]!;
+}
 
-    const tokenAccountInfo = await connection.getAccountInfo(ataAddress);
-    const tokenAccount = deserializeTokenAccount(tokenAccountInfo?.data);
-    if (tokenAccount) {
-      return { address: ataAddress, ...emptyInstruction };
-    }
+type ResolvedTokenAddressRequest = {
+  tokenMint: PublicKey;
+  wrappedSolAmountIn?: u64;
+};
 
-    const createAtaInstruction = Token.createAssociatedTokenAccountInstruction(
-      ASSOCIATED_TOKEN_PROGRAM_ID,
-      TOKEN_PROGRAM_ID,
-      tokenMint,
-      ataAddress,
-      ownerAddress,
-      ownerAddress
+/**
+ * IMPORTANT: wrappedSolAmountIn should only be used for input/source token that
+ *            could be SOL. This is because when SOL is the output, it is the end
+ *            destination, and thus does not need to be wrapped with an amount.
+ *
+ * @param connection Solana connection class
+ * @param ownerAddress The user's public key
+ * @param tokenMint Token mint address
+ * @param wrappedSolAmountIn Optional. Only use for input/source token that could be SOL
+ * @returns
+ */
+export async function resolveOrCreateATAs(
+  dal: OrcaDAL,
```

Comment:
given resolveOrCreateATA is a util class function usable by all of Orca, we may not want to use a whirlpool specific class (DAL) here. Can we just pass in the accountRentExempt value or a generic function to fetch the accountRentExempt value here?